### DISCUSSION TITLE
Custom miner

### DISCUSSION THREAD

**Discussion by bro-ee** [RainbowMiner/RainbowMiner#1718] [Q&A] [answered]
Hello,
There are two questions:

1) I have created custom file in Miners folder "xxx.ps1" for example
It was working before update, after update everything is removed
Miner files itself in Bin folder are not removed
So - how to keep custom *.ps1 file?

2) MRR sees rig that I manually created in MRR website. But it's not adding new rigs for available algos...?

**Comment by RainbowMiner** [comment, marked as answer]
Sorry, it took me some time to answer:

1. custom miners: there is a work-around for your problem. Put your custom miner in the folders `.\Miners` and `.\MinersOldVersions` - be careful to update always both versions. During the automatic update, the `.\Miners` folder will be reset. But - each miner, that has been in both folders before, will be copied from `.\MinersOldVersions` to `.\Miners` after the update.
2. is this problem resolved by now?
